### PR TITLE
z_collision_check_OK

### PR DESCRIPTION
**Pull request by RevoSucks**
I had to use a weird struct for one TriNorm, because 3 separate symbols kept being reordered, and the 2nd one was an extra +8 for some reason. I have no clue what IDO put there, so I kinda cludged it for now.

I also moved prevent_bss_reordering lower to make the first part of BSS correct. Its a workaround, but the file is correct now, well, maybe.

Feel free to suggest a better match.

### CODE REVIEW

**Comment by RevoSucks** [comment]
Whoops, 2 variables slipped into tools/variables.txt. Oh well, I'm tired. Whatever.

**Comment by RevoSucks** [comment]
Hold on, something broke. It was OK earlier.

**Comment by RevoSucks** [comment]
I fixed the weird struct. It was not needed after all.

**Comment by AngheloAlf** [comment]
fyi we have been avoiding this file because of how much bss it has. Header changes can affect very aggressively to the bss reorder, specially in in-function-static bss. We have been waiting to have the headers a bit more splitted before tackling this one, so we can not include `global.h` and have finer control over the included stuff.
Due to all of this, this PR will be in "held" hell for a long while, like #1450 is

File: tools/disasm/variables.txt [hunk 1]
```diff
@@ -2269,6 +2269,8 @@
     0x801D703C:("sOcarinaMemoryGameEndPos","UNK_TYPE1","",0x1),
     0x801D7040:("sOcarinaMemoryGameNumNotes","UNK_TYPE1","",0x1),
     0x801D7044:("sOcarinaSongNotes","OcarinaNote","[24][20]",0xF00),
+    0x801D7E04:("D_801D7E04","UNK_TYPE1","",0x1),
+    0x801D7EA4:("D_801D7EA4","UNK_TYPE1","",0x1),
```
Comment (hensldm) [inline tools/disasm/variables.txt:2273]: Can you revert this file back? Looks like it got added by accidently from a previously closed PR.

File: src/code/z_collision_check.c [hunk 2]
```diff
@@ -1271,6 +1250,10 @@ ColChkResetFunc sOCResetFuncs[] = {
     Collider_ResetQuadOC,   Collider_ResetSphereOC,
 };
 
+TriNorm D_801EE6C8;
+u8 D_801EE6FC_unused[4];
```
Comment (hensldm) [inline src/code/z_collision_check.c:1254]:
```suggestion
static s32 sBssPad;
```
What we have been calling unused bss variables. Though kind of weird this is needed here, but not with the other TriNorm pairs but if it works it works I guess🤷 .

**Review by hensldm — COMMENTED** [review]
Overall changes seem fine to me. Though as Anghelo said we definitely should let this sit through multiple merges to see how stable bss is.

File: tools/disasm/variables.txt (hunk 1 above)
Comment (RevoSucks) [reply tools/disasm/variables.txt]: reverted

File: src/code/z_collision_check.c (hunk 2 above)
Comment (RevoSucks) [reply src/code/z_collision_check.c]: edited

File: src/code/z_collision_check.c [hunk 3]
```diff
@@ -1271,6 +1250,10 @@ ColChkResetFunc sOCResetFuncs[] = {
     Collider_ResetQuadOC,   Collider_ResetSphereOC,
 };
 
+TriNorm D_801EE6C8;
+static s32 sBssPad;
```
Comment (hensldm) [inline src/code/z_collision_check.c:1254]: Hmm, just tested locally and this pad doesn't seem necessary?

Comment (RevoSucks) [reply src/code/z_collision_check.c]: oh. Thats weird. I wonder why IDO decided to align this one. Removed

**Review by hensldm — APPROVED** [review]
LTGM. Just would like to wait to check for stability.

**Review by engineer124 — APPROVED** [review]
Content looks good for me, I also agree on waiting to test for bss stability.

In particular, unlike https://github.com/zeldaret/mm/pull/1450 where we've very carefully removed all header dependencies on anything that's likely to change, this file is still connected to `#include "global.h"` due to `PlayState`.

I'd preferably like to see weekly merges of main into this PR, but I understand that's high maintenance. Is this something you'd like to stay on top of or would you like help keeping this branch frequently up-to-date?

**Comment by RevoSucks** [comment]
I will periodically update it, yes.

**Comment by starlightgemini214** [comment]
..syug em pleh esaelP ?emag siht ni devomer saw ti ebyam rO .kcattA nipS fo edargpu mumixam eht tuohtiw ksaM s'arojaM ni elpmet tsal eht retne ot desoppus m'I fi desufnoc m'I

**Comment by altoidmerope294** [comment]
I'v heared u can but can u?

**Comment by hirottau302** [comment]
.htob ni detseretni ma I tub ,retteb hcum s'ti dlot neeb evah dna ,SD3 eht evah I

**Comment by hirottau302** [comment]
I got the mirror shield and I beat the stone sity place but I dont know where the temple entrance is.

**Comment by hirottau302** [comment]
.emag eht fo mhtyhr eht otni kcab gnitteg fo yaw a sa htiw trats ot tseuq tseisae eht tnaw tsuj I

**Comment by yemenalcyone230** [comment]
elpmeT yaB taerG eht ot teg 
ot anopE gnitteG

**Comment by ginomeelectra202** [comment]
?emag eht ecnavda nac I taht os niaga revo lla ssob eht taeb ot evah I oD ?neppah ot desoppus siht sI .ereht llits saw ssob elpmet eht dna detlem ton saw eci eht ,emag devas ym ot kcab emac I nehw tub ,emit fo gnos eht gnisu devas neht I ;)eci eht detlem neht dna( elpmet daeh wons/daeh wons fo level eritne eht hguorht deyalp I ,ecnatsni roF .detcelloc smeti eht ylno ,emit fo gnos eht htiw devas ton si ssergorp ym yhw dnatsrednu t'nod I ,revewoh ;)emit fo gnos eht gniyalp yb eht dna ,seutats lwo eht htiw selif yraropmet .e.i( evas ot syaw owt era ereht ezilaer I

**Comment by altoidmerope294** [comment]
There's one turn that i can't get right. (second turn that dosent have chests)

**Comment by hensldm** [comment]
Would you like help updating this PR? Still would like to see it go through some more merges, but would be good to see it updated to see how bss treats us.

**Pull request merged**